Rework communication docs, add Vault tutorial (#82)
infracloudio/botkube-docs: content/configuration/communication/_index.md

---
menutitle: "Communication"
title: "Communication"
weight: 30
---

BotKube backend reads communication settings from two sources:
- the [comm_config.yaml](#file-syntax) file placed in the directory specified by the **CONFIG_PATH** environment variable,

> - the exported [environment variables](#environment-variables). It takes priority over the configuration specified in the file.

Reviewer: ```suggestion
- the exported [environment variables](#environment-variables). They take priority and override the configuration specified in the file.

```

## File syntax

The communication configuration file contains:

> - communication mediums configuration,

Reviewer: Maybe `platforms` is a better word? 🤔 

BTW webhook / Elasticsearch doesn't fit that nicely into both `mediums` and `platforms` I think 🤔 But I don't have any better word for that 

```suggestion
- communication platforms configuration,

```

Author: It was there already but I'm ok with the change 👍

      awsRegion: 'us-east-1'            # AWS region where Elasticsearch is deployed
      roleArn: ''                       # AWS IAM Role arn to assume for credentials,
                                        # use this only if you dont want to
                                        # use the EC2 instance role or not
                                        # running on AWS instance
    server: 'ELASTICSEARCH_ADDRESS'     # e.g https://example.com:9243
    username: 'ELASTICSEARCH_USERNAME'  # Set creds if using Basic Auth
    password: 'ELASTICSEARCH_PASSWORD'
    # ELS index settings
    index:
      name: botkube
      type: botkube-event
      shards: 1
      replicas: 0

  # Settings for Webhook
  webhook:
    enabled: false
    url: 'WEBHOOK_URL'                 # e.g https://example.com:80
```

The default configuration can be found at:
- for Helm chart in [values.yaml](https://github.com/infracloudio/botkube/blob/master/helm/botkube/values.yaml),
- and for all-in-one deployment spec in [deploy-all-in-one.yaml](https://github.com/infracloudio/botkube/blob/master/deploy-all-in-one.yaml).

## Environment variables

> The individual communication settings can be specified via environment variables. It takes priority over the configuration specified in the file.

Reviewer: ```suggestion
The individual communication settings can be specified via environment variables. They take priority and override the configuration specified in the file.

```

To construct the environment variable name, take any property from the [configuration file](#file-syntax) and make it uppercase. Use the underscore for properties that are nested. That's all - you have the environment variable name!

For example, such configuration property from YAML:
```yaml
communications:
  slack:
    token: 'SLACK_API_TOKEN'
```

> is represented by `COMMUNICATIONS_SLACK_TOKEN` environment variable.

Reviewer: Not sure about the word `represented` 🤔 
I would use something like `And this is the corresponding environment variable: `COMMUNICATIONS_SLACK_TOKEN`.

> This is a useful feature that allows you to store the overall configuration in a file, where sensitive data, such as tokens, can be put in environment variables. See the [**Tokens from Vault via CSI driver**](/configuration/communication/vault-csi/) tutorial for the example use-case.

Reviewer: ```suggestion
This is a useful feature that allows you to store the overall configuration in a file, where sensitive data, such as tokens, can be put in environment variables. See the [**Tokens from Vault via CSI driver**](/configuration/communication/vault-csi/) tutorial for an example use-case.

```
